CIP-0022/CIP-0022.md

```diff
@@ -0,0 +1,134 @@
+---
+CIP: 22
+Title: Pool operator verification
+Authors: Andrew Westberg <andrewwestberg@gmail.com>, Martin Lang <martin@martinlang.at>, Ola Ahlman <ola@ahlnet.nu>
+Comments-URI: no comments yet
+Status: Draft
+Type: Informational
+Created: 2021-06-21
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This proposal describes a method allowing a stakepool operator to provide credentials to verify that they are the rightful manager for their stakepool.
+
+# Motivation
+
+Many websites such pooltool.io, adapools.org, and others need to allow pool operators special access to modify the way their pool appears on the website. SPOCRA and other organizations also have a need to allow voting on proposals and ensure that each vote cast is from a valid pool operator. Today, these sites and organizations all use different techniques for validating pool operators.
+
+pooltool.io - Validates operators by receiving 1 ada spent from the pool's registered rewards account
+
+adapools.org - Validates operators by requesting that the operator include a special generated value in their extended pool metadata json file.
+
+This proposal desires to simplify and streamline a single approach that all can reference in order to verify that a pool operator is who they say they are.
+
+Current Tools that have implemented it already or have plans to implement it:
+* [CNCLI](https://github.com/AndrewWestberg/cncli)
+* [JorManager](https://bitbucket.org/muamw10/jormanager/)
+* [StakePoolOperator Scripts](https://github.com/gitmachtl/scripts)
+* [CNTools](https://cardano-community.github.io/guild-operators/#/Scripts/cntools)
+
+
+# Specification
+
+In order to achieve the goals of this CIP, the pool operator needs to provide some credential or credentials to the validating party which cannot be spoofed. The VRF pool keys and VRF signature algorithm implemented in libsodium are chosen to build and provide this credential/signature. This signature can then be validated and the operator verified without ever exposing any of the pool's private key information. This technique is very similar to verifying that a block produced by another pool is valid. The only difference is that instead of validating the slot seed for a given pool, we're validating a pre-determined message hash.
+
+## Verification Steps:
+
+1. Stakepool Operator (SPO) sends their pool_id to the Validating Server (VS)
+2. VS checks the blockchain to retrieve the most recent block made by the pool. If the pool has made a block, the VRF public key is already known to the VS.
+3. The VS sends a challenge request to the SPO. 
+    1. VS creates a blake2b hash of "cip-0022{seedphrase}{random_nonce}"
```

Comment:
It might make sense maybe to maintain a small registry.json as part of this CIP with registered seedphrases so that we reduces chances of clashes across services.

Or alternatively, force the seed phrase to be the host name of the service the SPO is authenticating against. This fixes quite many issues / attacks of type man-in-the-middle.

Reply:
Updated to require validating server to use their domain name as the seedphrase.